### ISSUE TITLE
Bundle is now VALID, then INVALID, and then VALID again

### ISSUE THREAD

**Issue by rodoabad** [webpack/karma-webpack#92]
CLI output.

```
[13:41:50] Starting 'karma'...
[13:41:50] Finished 'karma' after 123 ms
Hash: 8344a6c0a9b3c44a5636
Version: webpack 1.12.4
Time: 10ms
webpack: bundle is now VALID.
webpack: bundle is now INVALID.
Hash: 57a7c9b4fe1d9e5d76e3
Version: webpack 1.12.4
Time: 2006ms
           Asset     Size  Chunks             Chunk Names
webpack.karma.js  3.57 MB       0  [emitted]  webpack.karma.js
chunk    {0} webpack.karma.js (webpack.karma.js) 1.36 MB [rendered]
    [0] ./webpack.karma.js 930 bytes {0} [built]
    [1] ./~/angular/index.js 48 bytes {0} [built]
    [2] ./~/angular/angular.js 1.06 MB {0} [built]
    [3] ./~/angular-mocks/angular-mocks.js 82.8 kB {0} [built]
    [4] ./src/client/app.js 2.57 kB {0} [built]
    [5] ./~/angular-ui-router/release/angular-ui-router.js 163 kB {0} [built]
    [6] ./src/client/config.js 784 bytes {0} [built]
    [7] ./src/client/views/helloWorld/index.js 1.1 kB {0} [built]
    [8] ./src/client/views/helloWorld/config.js 1.4 kB {0} [built]
    [9] ./src/client/views/petList/index.js 2.29 kB {0} [built]
   [10] ./src/client/views/petList/config.js 2.15 kB {0} [built]
   [11] ./src/client/views/petList/controller.js 970 bytes {0} [built]
   [12] ./src/client/views/petList/service.js 3 kB {0} [built]
   [13] ./src/client/views/petList/index.html 242 bytes {0} [built]
   [14] ./src/client/views/root/index.js 1.71 kB {0} [built]
   [15] ./src/client/views/root/config.js 1.37 kB {0} [built]
   [16] ./src/client/views/root/index.html 375 bytes {0} [built]
   [17] ./src/client/views/user/index.js 2.11 kB {0} [built]
   [18] ./src/client/views/user/config.js 1.59 kB {0} [built]
   [19] ./src/client/views/user/controller.js 590 bytes {0} [built]
   [20] ./src/client/views/user/index.html 619 bytes {0} [built]
   [21] ./test \.spec\.js 428 bytes {0} [built]
   [22] ./test/helloWorld/hello-world-route.spec.js 4.05 kB {0} [optional] [built]
   [23] ./test/petList/pet-list-ctrl.spec.js 284 bytes {0} [optional] [built]
   [24] ./test/petList/pet-list-route.spec.js 7.31 kB {0} [optional] [built]
   [25] ./test/petList/pet-list-svc.spec.js 5.91 kB {0} [optional] [built]
   [26] ./test/root/main-container.spec.js 3.58 kB {0} [optional] [built]
   [27] ./test/root/root.spec.js 3.28 kB {0} [optional] [built]
   [28] ./test/user/user-info.spec.js 4.32 kB {0} [optional] [built]
   [29] ./test/user/user-route.spec.js 3.87 kB {0} [optional] [built]
webpack: bundle is now VALID.
............
PhantomJS 1.9.8 (Windows 8 0.0.0): Executed 12 of 12 SUCCESS (0.073 secs / 0.017 secs)

Process finished with exit code 0
```

Not really sure what's happening here.

Here's my karma config.

```javascript
module.exports = function (config) {

  config.set({
    basePath: '',
    frameworks: [
      'chai',
      'mocha',
      'sinon'
    ],
    files: [
      'webpack.karma.js'
    ],
    preprocessors: {
      'webpack.karma.js': [
        'sourcemap',
        'webpack'
      ]
    },
    webpack: {
      devtool: 'inline-source-map',
      module: {
        loaders: [
          {
            test: /\.js$/,
            exclude: /node_modules/,
            loaders: [
              'ng-annotate',
              'babel-loader'
            ]
          },
          {
            test: /\.html$/,
            loader: 'html'
          }
        ]
      }
    },
    exclude: [],
    reporters: [
      'dots',
      //'coverage',
      //'coveralls'
    ],
    port: 9876,
    colors: true,
    logLevel: config.LOG_ERROR,
    autoWatch: false,
    browsers: [
      'PhantomJS'
    ],
    singleRun: true,
    coverageReporter: {
      reporters: [
        {
          type: 'lcov',
          dir: 'reports/coverage',
          subdir: '.'
        }
      ]
    }
  });

};
```

**Comment by thomasklein** [comment]
@rodoabad how did you solve it?

**Comment by antwan1986** [comment]
Hey @rodoabad I'd like to know how you solved this too; I'm getting this currently.

**Comment by amercier** [comment]
@rodoabad I'm having the same issue. I'd like to know how you solved this too.

**Comment by amercier** [comment]
I ended up suppressing all non-errors messages:

```
    webpack: {
       ...
    },
    webpackServer: {
      noInfo: true // Suppress all webpack messages, except errors
    }
```

**Comment by Dekeyras** [comment]
Did you ever figure this out? I have just noticed this "valid, invalid, value" thing happening in our build too

**Comment by stachuuu93** [comment]
Try to add
`
webpackMiddleware: {
      stats: 'errors-only',
      noInfo: true
 },
` 
This works for me.

**Comment by duyhoang** [comment]
Do you use VS Code? If so, close VS Code and see what happens.

**Comment by Krisa** [comment]
@duyhoang hard to believe but... you're right, that was the problem in my case. Any idea on how to fix this? (not using VS Code is not really an option :) )

**Comment by WaleedAshraf** [comment]
I also have same issue, but it only happens some time. Not sure whats main cause ? Anyone figured it out?

**Comment by bpossolo** [comment]
I've noticed VSCode trigger bundle rebuilds as well....
Happens when I simply open VSCode and then continues to happen without me even touching VSCode.

**Comment by Krisa** [comment]
@bpossolo - That's actually because you must have Git setup with VSCode, and VSCode polls regularly the remote, to check for any change. What happens is that the `HEAD` file (or something similar) in the `.git` folder is updated and, additionally, your current webpack setup is watching for every change within your root folder... including the `.git` folder. Basically, you need to tweak your setup, to ignore somehow `.git`.

**Comment by bpossolo** [comment]
Thank you for this. Yes you're right VS Code is aware of git and detecting local changes so it might be fetching remote changes. I'll investigate tomorrow!.

**Comment by bpossolo** [comment]
@Krisa  thanks for the tip.
Turns out webpack-dev-server was watching the root project directory for changes and all files that participate in the build.
VSCode was periodically touching the .git folder like you said.
I used the WebpackWatchIgnore plugin to turn off watching the root directory and also node_modules.
I tried to ignore .git specifically but this didnt work because webpack-dev-server is watching the root folder (which contains `.git/`) rather than watching `.git/` itself

Anyways, it now watches all our source files properly and doesn't constantly rebuild!!
```
const projectRootDir = path.resolve(__dirname, '../..'); // this config file is two folders below project root
const watchIgnoreList: RegExp[] = [
  new RegExp(projectRootDir + '$'),
  /node_modules/
];

webpackConfig = {
  ...
  plugins: [
    new webpack.WatchIgnorePlugin(watchIgnoreList)
  ]
}
```
